### PR TITLE
User can buy a subscription

### PR DESCRIPTION
**Pull request by johanperjulius1**
## This PR contains:

- Feature tests
- Refactoring tests by adding Cypress commands
- Adding login functionality with J-tock-auth
- Payment form
- Payment functionality with stripe
- Refactoring with Redux

## [PT](https://www.pivotaltracker.com/story/show/175113657)

### CODE REVIEW

**Review by FacundoOs — APPROVED** [review]

File: cypress/integration/userCanViewPartOfPremiumContent.feature.js [hunk 1]
```diff
@@ -41,5 +70,38 @@ describe("Premium article view for visitor/user", () => {
...
+
+  context("subscriber", () => {
+    beforeEach(() => {
+      cy.login("subscriber");
+      cy.get("[data-cy='article-1']").within(() => {
+        cy.get("[data-cy='title']").click();
+      });
+    });
+
+    it("can only see first 300 characters", () => {
```
Comment (emtalen) [inline cypress/integration/userCanViewPartOfPremiumContent.feature.js:91]: Why can a subscriber only see the first 300 characters?

File: src/index.css [hunk 2]
```diff
@@ -40,3 +40,10 @@
 #weather {
   background-color: white;
 }
+
+.centered {
+  max-width: 60%;
+  margin: auto
+  /* margin-left: 60px; */
```
Comment (emtalen) [inline src/index.css:47]: commented out code

File: src/index.js [hunk 3]
```diff
@@ -16,9 +23,13 @@ if (process.env.NODE_ENV === "production") {
 axios.defaults.baseURL = apiUrl;
 
 ReactDOM.render(
-  <BrowserRouter>
-    <App />
-  </BrowserRouter>,
+  <StripeProvider apiKey="pk_test_21nBNjeqdyB1Mzm2VjDPQprF00kyEKYZSK">
```
Comment (emtalen) [inline src/index.js:26]: Put in .env file

File: src/index.js [hunk 4]
```diff
@@ -16,9 +23,13 @@ if (process.env.NODE_ENV === "production") {
 axios.defaults.baseURL = apiUrl;
```
Comment (emtalen) [inline src/index.js:23]: Put the api urls in .env file

**Pull request merged**